### ISSUE TITLE
TagFolder slows down / halts Obsidian

### ISSUE THREAD

**Issue by gala8y** [vrtmrz/obsidian-tagfolder#19]
Thx for developing TagFolder. It is closest to what I really would like to see in Obsidian (see my FR at https://forum.obsidian.md/t/filter-tags-list-in-tag-pane-when-choosing-tags/12352).

However, TagFolder brings Obsidian to stop for me on a vault with ~ 5k notes.
If i can provide any information to help maybe see why it is so, I'd be happy to send it in.

Thx.

**Comment by vrtmrz** [comment]
Thank you for your report! 
Wow, 5k note was not supposed yet, but my vault is also growing. It's our problem. 

I’ll try to reproduce this once, but perhaps I ask so.

**Comment by vrtmrz** [comment]
@gala8y 
Reproduced and fixed! 
Could you please try v0.10.1?

**Comment by gala8y** [comment]
Hi @vrtmrz, thank you for looking into it. Sorry for late reply, I was AFK.

I tried to test it a little bit and I still see substential lag (delayed typing, moving panes, overall response from the app) when TagFolder pane is left open/visible. When I switch to other Pane in a sidebar (file explorer, search, anything) effect disappers.
I think the reason is that there are so many items in the viewport to style/render when TagFolder pane is visible.

Now I removed few big files with many tags inside them, which I brought to Obsidian from earlier system and did not process (chunked down). This seems to make things much better, as TagFolder does not get populated with long branches of many tags pertaining to only one file. The lag still persists a bit, but it is much better.

Also, is "Tag scanning delay" in seconds, minutes?

Thx.
